timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -902,9 +903,61 @@ template <typename RealType, bool Interpolated> void declare_nonbonded(py::modul
+        .def(
+            py::init([](const py::array_t<int, py::array::c_style> &lambda_plane_idxs_i,  //
+                        const py::array_t<int, py::array::c_style> &lambda_offset_idxs_i, //
+                        const double beta,
+                        const double cutoff,
+                        const std::string &transform_lambda_charge = "lambda",
+                        const std::string &transform_lambda_sigma = "lambda",
+                        const std::string &transform_lambda_epsilon = "lambda",
+                        const std::string &transform_lambda_w = "lambda") {
+                std::vector<int> lambda_plane_idxs(lambda_plane_idxs_i.size());
+                std::memcpy(
+                    lambda_plane_idxs.data(), lambda_plane_idxs_i.data(), lambda_plane_idxs_i.size() * sizeof(int));
+
+                std::vector<int> lambda_offset_idxs(lambda_offset_idxs_i.size());
+                std::memcpy(
+                    lambda_offset_idxs.data(), lambda_offset_idxs_i.data(), lambda_offset_idxs_i.size() * sizeof(int));
+
+                std::string dir_path = dirname(__FILE__);
+                std::string kernel_dir = dir_path + "/kernels";
+                std::string src_path = kernel_dir + "/k_lambda_transformer_jit.cuh";
+                std::ifstream t(src_path);
+                std::string source_str((std::istreambuf_iterator<char>(t)), std::istreambuf_iterator<char>());
+                source_str = std::regex_replace(source_str, std::regex("KERNEL_DIR"), kernel_dir);
+                source_str =
+                    std::regex_replace(source_str, std::regex("CUSTOM_EXPRESSION_CHARGE"), transform_lambda_charge);
+                source_str =
+                    std::regex_replace(source_str, std::regex("CUSTOM_EXPRESSION_SIGMA"), transform_lambda_sigma);
+                source_str =
+                    std::regex_replace(source_str, std::regex("CUSTOM_EXPRESSION_EPSILON"), transform_lambda_epsilon);
+                source_str = std::regex_replace(source_str, std::regex("CUSTOM_EXPRESSION_W"), transform_lambda_w);
+
+                return new timemachine::NonbondedDense<RealType, Interpolated>(
+                    lambda_plane_idxs, lambda_offset_idxs, beta, cutoff, source_str);
+            }),
+            py::arg("lambda_plane_idxs_i"),
+            py::arg("lambda_offset_idxs_i"),
+            py::arg("beta"),
+            py::arg("cutoff"),
+            py::arg("transform_lambda_charge") = "lambda",
+            py::arg("transform_lambda_sigma") = "lambda",
+            py::arg("transform_lambda_epsilon") = "lambda",
+            py::arg("transform_lambda_w") = "lambda");
+}
+
 template <typename RealType, bool Interpolated> void declare_nonbonded_pairs(py::module &m, const char *typestr) {
 
-    using Class = timemachine::NonbondedPairs<RealType, Interpolated>;
+    using Class = timemachine::NonbondedPairs<RealType, false, Interpolated>;
```

Comment:
Nice -- I think this makes sense to keep the wrapper specialized to `Negated=false`

Reply:
Agreed -- I considered exposing this to the Python wrapper, but it seemed best to only wrap the non-negated version for now for simplicity.